resources/ubuntu-build.sh

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env bash
+
+set -x
+set -e
+
+USRSCTPPATH=usrsctp
+JNIPATH=src/main/native
+JAVAHPATH=target/native/javah
+OUTPATH=target/classes/lib
+RESOURCESPATH=src/main/resources
+
+if [ "$#" -ne 3 ]; then
+    echo "Usage: $0 <ARCH> <JAVA_VERSION> <DIR>"
+    echo "  ARCH: Architecture to build for (amd64, arm64, ppc64el)"
+    echo "  JAVA_VERSION: Java version (11)"
+    echo "  DIR: sctp4j project directory"
+    exit 1
+fi
+
+DEBARCH=$1
+JAVA_VERSION=$2
+DIR="$3"
+
+cd "$DIR"
+
+if [ \! -r $JAVAHPATH/org_jitsi_modified_sctp4j_SctpJni.h ]; then
+    echo "$JAVAHPATH/org_jitsi_modified_sctp4j_SctpJni.h not found: did you run mvn compile?"
+    exit 1
+fi
+
+case $DEBARCH in
+    amd64)
+	GNUARCH=x86_64
+	JNAARCH=x86-64
+	;;
+    arm64)
+	GNUARCH=aarch64
+	JNAARCH=aarch64
+	;;
+    ppc64el)
+	GNUARCH=powerpc64le
+	JNAARCH=ppc64le
+	;;
+    *)
+	echo "ERROR: Unsupported arch $DEBARCH!"
+	exit 1
+	;;
+esac
+
+NATIVEDEBARCH=$(dpkg --print-architecture)
+
+if [ $DEBARCH != $NATIVEDEBARCH ]
+then
+    HOST=$GNUARCH-linux-gnu
+    CONFIGURE_ARGS="--host $HOST"
+    export CC=$HOST-gcc
+else
+    export CC=gcc
+fi
+
+NCPU=$(grep -c processor /proc/cpuinfo)
+if [ -n "$NCPU" -a "$NCPU" -gt 1 ]
+then
+    MAKE_ARGS="-j $NCPU"
+fi
+
+cd $USRSCTPPATH
+./bootstrap
+
+if [ -r Makefile ]; then
+    make distclean
+fi
+
+OBJ_DIR=obj-$DEBARCH
+INSTALL_DIR=$DIR/$USRSCTPPATH/install-$DEBARCH
+rm -rf $OBJ_DIR $INSTALL_DIR
+mkdir $OBJ_DIR
+cd $OBJ_DIR
+../configure --with-pic --enable-invariants --prefix=$INSTALL_DIR $CONFIGURE_ARGS
+make $MAKE_ARGS install
+
+export JAVA_HOME=/usr/lib/jvm/java-$JAVA_VERSION-openjdk-$NATIVEDEBARCH
+echo $JAVA_HOME
+java -version
+
+cd "$DIR"
+mkdir -p $OUTPATH
+cd $OUTPATH
+SO_DIR=linux-$JNAARCH
+rm -rf $SO_DIR
+mkdir $SO_DIR
+cd $DIR
+$CC -c -g -fPIC -std=c99 -O2 -Wall \
```

Comment:
Perhaps we should have a Makefile somewhere?

Reply:
Or use CMake, see e.g. srtp for an example.

Reply:
I'm not too familiar with CMake, but I can play with it.

Reply:
A plain Makefile would probably be annoying with the differences between Mac and Linux?

Reply:
Not sure if there would be many differences, but you'll loose the advantage that CMake would provide nice support for IDEs.